### ISSUE TITLE
[RFE] Disallow spaces in the command parameter

### ISSUE THREAD

**Issue by Fryguy** [ManageIQ/awesome_spawn#59]
This would enforce the usage of the :params when needed, and a user would _not_ be able to bypass it.  If they are going to bypass anyway, then there is no point in using AwesomeSpawn.

The only use case we _might_ have to account for is `sudo`.  I saw might, because one case easily do

```ruby
AwesomeSpawn.run!("sudo", :params => ["echo", "'hi'")
```

But that is kind of weird.  I think I'd prefer another param:

```ruby
AwesomeSpawn.run!("echo", :params => "'hi'", :sudo => true)
AwesomeSpawn.run!("echo", :params => "'hi'", :sudo => "someuser")
```

Which would be nice because a caller could build the options and call it, then on failure just tack in another option (instead of having to re-call AwesomeSpawn and reorder all of the params.

**Comment by chessbyte** [comment]
yeah, I like `:sudo` as another parameter, because it is *not* the actual command
